scapy/utils.py

```diff
@@ -1253,35 +1278,48 @@ def _write_header(self, pkt):
         self.f.flush()
 
     def write(self, pkt):
-        """accepts either a single packet or a list of packets to be
-        written to the dumpfile
+        """
+        Writes a Packet or bytes to a pcap file.
 
+        :param pkt: Packet(s) to write (one record for each Packet), or raw
+                    bytes to write (as one record).
+        :type pkt: iterable[Packet], Packet or bytes
         """
         if isinstance(pkt, bytes):
             if not self.header_present:
                 self._write_header(pkt)
             self._write_packet(pkt)
         else:
             pkt = pkt.__iter__()
-            if not self.header_present:
-                try:
-                    p = next(pkt)
-                except (StopIteration, RuntimeError):
-                    self._write_header(None)
-                    return
-                self._write_header(p)
-                self._write_packet(p)
             for p in pkt:
+                if not self.header_present:
+                    self._write_header(p)
                 self._write_packet(p)
 
-    def _write_packet(self, packet, sec=None, usec=None, caplen=None, wirelen=None):  # noqa: E501
-        """writes a single packet to the pcap file
+    def _write_packet(self, packet, sec=None, usec=None, caplen=None,
+                      wirelen=None):
+        """
+        Writes a single packet to the pcap file.
+
+        :param packet: bytes for a single packet
+        :type packet: bytes
+        :param sec: time the packet was captured, in seconds since epoch. If
+                    not supplied, defaults to now.
+        :type sec: int or long
+        :param usec: If ``nano=True``, then number of nanoseconds after the
+                     second that the packet was captured. If ``nano=False``,
+                     then the number of microseconds after the second the
+                     packet was captured
+        :type usec: int or long
+        :param caplen: The length of the packet in the capture file. If not
+                       specified, uses ``len(packet)``.
+        :type caplen: int
+        :param wirelen: The length of the packet on the wire. If not
+                        specified, uses ``caplen``.
+        :type wirelen: int
+        :returns: None
+        :rtype: None
         """
-        if isinstance(packet, tuple):
-            for pkt in packet:
-                self._write_packet(pkt, sec=sec, usec=usec, caplen=caplen,
-                                   wirelen=wirelen)
-            return
```

Comment:
I'll agree that tuple support is a weird thing we were doing. Though I'm not sure why it was there in the first place, that's a backward incompatible change.

I'll check it out to see how used it was